packages/nimble-components/src/table-column/base/template.ts

```diff
@@ -1,10 +1,22 @@
-import { html } from '@microsoft/fast-element';
+import { html, ref, slotted } from '@microsoft/fast-element';
 import type { TableColumn } from '.';
 
+// prettier-ignore
 export const template = html<TableColumn>`
     <template slot="${x => x.columnInternals.uniqueId}">
-        <span class="header-content">
-            <slot></slot>
+        <span
+            ${ref('headerSpan')}
+            class="header-content"
+            @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.slottedHeaderContent.map(element => element.textContent?.trim()).join(' ')
+                    && x.headerSpan.offsetWidth < x.headerSpan.scrollWidth;
+    }}"
+            @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+            title=${x => (x.isValidContentAndHasOverflow ? x.slottedHeaderContent.map(element => element.textContent?.trim()).join(' ') : null)}
+        >
+            <slot ${slotted('slottedHeaderContent')}></slot>
```

Comment:
The overhead of persisting and tracking reference to the slotted content isn't needed.
Can you make a ref to the slot and use it instead in the method above being added
```ts
contentSlot.assignedNodes().map(...)
```

Reply:
Done